core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1194,6 +1216,9 @@ object GroupMetadataManager {
       if (version > 0)
         memberStruct.set(REBALANCE_TIMEOUT_KEY, memberMetadata.rebalanceTimeoutMs)
 
+      if (version >= 3)
+        memberStruct.set(GROUP_INSTANCE_ID_KEY, memberMetadata.groupInstanceId)
```

Comment:
I'm trying to think through the implications of this. We are silently discarding the instance id which means that replicas won't know about it. The member will be considered a static member until there is a coordinator change. Then it will suddenly become dynamic again and I think that would trigger this assertion: https://github.com/apache/kafka/pull/6177/files#diff-a210b6f573f800e8a562fda9d53cd148R269. I think we probably need to avoid using the static membership logic entirely until the IBP supports it.

Reply:
That sounds reasonable. I'm not sure I'm fully following here because unless broker upgrades to latest, the group instance id should not include the join group because of automatic request downgrade.

Reply:
What's `IBP`?

Reply:
@guozhangwang inter broker protocol

Reply:
Addressed in GroupCoordinator.scala